source/base/data_out_base.cc

```diff
@@ -5302,6 +5304,26 @@ namespace DataOutBase
 
 
 
+  namespace internal
+  {
+    template <int dim>
+    TableIndices<2>
+    unrolled_to_component_indices(const unsigned int i)
```

Comment:
That's a bad name because it's not clear whether you want to transfer between vector, rank-2 tensor, etc objects and plain index space. Please use the corresponding functions directly from the tensor classes.

Reply:
> Please use the corresponding functions directly from the tensor classes.

I explicitly do NOT use it directly as Tensor2 does not like `dim=2` and there is a compiler warning of division by zero. Otherwise I would never introduce this wrapper in the first place.

Reply:
I assumed you had good reasons to do things this way, but let's try to find ways to make this work somehow. What is the error message you are getting?

Reply:
```
      indices[r] = (remainder % dim);
      remainder /= dim;
```
in Tensor are not nice for `dim==0`. DataOut is instantiated for dim==0.

Reply:
Could we specialize this in the tensor class?

If you can't figure this out easily, I'd be ok if you renamed the current function to `rank_2_unrolled_...` and opened a github issue that reminds us that this could be cleaned up later on.